Core/Core/Search/Models/SearchDescriptor.swift

```diff
@@ -0,0 +1,33 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+import Combine
+
+public protocol SearchDescriptor {
+    associatedtype Filter
+    associatedtype FilterEditor: View
+    associatedtype Support: SearchSupportAction
+    associatedtype Display: View
+
+    var support: SearchSupportOption<Support>? { get }
+    var isEnabled: AnyPublisher<Bool, Never> { get }
+
+    func searchDisplayView(_ filter: Binding<Filter?>) -> Display
```

Comment:
I am not sure about what you mean here.
The reasoning behind naming this as Display, and is to avoid making the meaning of "Results" over bloated for this context.